File: leetcode/java/0118_pascals_triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<List<Integer>>(List.of(List.of(1), List.of(1, 1)));

        if (numRows == 1) return new ArrayList<List<Integer>>(List.of(List.of(1)));
        else if (numRows == 2) return res;
        else {
            for (int i = 2; i < numRows; i++) {
                List<Integer> subRes = new ArrayList<Integer>();
                for (int j = 0; j <= i; j++) {
                    if (j == 0 || j == i) subRes.add(1);
                    else subRes.add(res.get(i - 1).get(j - 1) + res.get(i - 1).get(j));
                }
                res.add(subRes);
            }
        }

        return res;
    }
}

File: leetcode/cpp/0118_pascals_triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ivec;

        for (int i = 0; i < numRows; i++) {
            ivec.push_back(vector<int>());

            for (int j = 0; j < i + 1; j++) {
                if (j == 0 || j == i) {
                    ivec[i].push_back(1);
                }
                else if (j != 0 && j != i) {
                    int num = ivec[i - 1][j - 1] + ivec[i - 1][j];

                    ivec[i].push_back(num);
                }
            }
        }

        return ivec;
    }
};